File: apps/frontend/src/lib/api/sec-filings.ts
/**
 * SEC Filings API Client
 * Client for SEC Edgar API endpoints - company filings, insider trading, annual/quarterly reports
 */

import { apiClient } from './client'
import type {
  SECFiling,
  SECFilingList,
  AnnualReport,
  QuarterlyReport,
  CurrentReport,
  InsiderTrade,
  FilingsSummary,
  CompanyInfoSEC,
} from '@/lib/types/sec-filings'

export const secFilingsApi = {
  // Company Info
  getCompanyInfo: (symbol: string) =>
    apiClient.get<CompanyInfoSEC>(`/api/v1/sec-filings/company/${symbol}`),

  // All Filings
  getCompanyFilings: (symbol: string, count = 10) =>
    apiClient.get<SECFilingList>(`/api/v1/sec-filings/filings/${symbol}`, {
      params: { count }
    }),

  // Annual Reports (10-K)
  getAnnualReports: (symbol: string, count = 5) =>
    apiClient.get<AnnualReport[]>(`/api/v1/sec-filings/10k/${symbol}`, {
      params: { count }
    }),

  // Quarterly Reports (10-Q)
  getQuarterlyReports: (symbol: string, count = 5) =>
    apiClient.get<QuarterlyReport[]>(`/api/v1/sec-filings/10q/${symbol}`, {
      params: { count }
    }),

  // Current Reports (8-K)
  getCurrentReports: (symbol: string, count = 10) =>
    apiClient.get<CurrentReport[]>(`/api/v1/sec-filings/8k/${symbol}`, {
      params: { count }
    }),

  // Insider Transactions
  getInsiderTransactions: (symbol: string, count = 50) =>
    apiClient.get<InsiderTrade[]>(`/api/v1/sec-filings/insider/${symbol}`, {
      params: { count }
    }),

  // Filings Summary
  getFilingsSummary: (symbol: string) =>
    apiClient.get<FilingsSummary>(`/api/v1/sec-filings/summary/${symbol}`),

  // Proxy Statements (DEF 14A)
  getProxyStatements: (symbol: string, count = 5) =>
    apiClient.get<SECFiling[]>(`/api/v1/sec-filings/proxy/${symbol}`, {
      params: { count }
    }),

  // Registration Statements (S-1, S-3)
  getRegistrationStatements: (symbol: string, count = 5) =>
    apiClient.get<SECFiling[]>(`/api/v1/sec-filings/registration/${symbol}`, {
      params: { count }
    }),
}

File: apps/backend/src/api/sec_filings.py
"""
SEC Filings API Router
Exposes SEC Edgar endpoints for frontend integration - company filings, insider trading, annual/quarterly reports
"""

from ninja import Router
from ninja_jwt import authenticate
from investments.tasks.sec_edgar_tasks import (
    fetch_company_info_sec,
    fetch_company_filings_sec,
    fetch_all_filings_sec,
    fetch_insider_transactions_sec,
    fetch_annual_reports_sec,
    fetch_quarterly_reports_sec,
    fetch_current_reports_sec,
    fetch_filings_summary_sec,
)

router = Router(tags=["SEC Filings"])


@router.get("/company/{symbol}")
@authenticate
def get_company_info(request, symbol: str):
    """Get SEC company information (CIK, SIC, state of incorporation)"""
    result = fetch_company_info_sec.delay(symbol)
    return result.get(timeout=30)


@router.get("/filings/{symbol}")
@authenticate
def get_company_filings(request, symbol: str, filing_type: str = None, count: int = 10):
    """Get all SEC filings for a company"""
    result = fetch_company_filings_sec.delay(symbol, count)
    return result.get(timeout=30)


@router.get("/summary/{symbol}")
@authenticate
def get_filings_summary(request, symbol: str):
    """Get filing type summary for a company"""
    result = fetch_filings_summary_sec.delay(symbol)
    return result.get(timeout=30)


@router.get("/10k/{symbol}")
@authenticate
def get_annual_reports(request, symbol: str, count: int = 5):
    """Get annual reports (10-K, 20-F, 40-F)"""
    result = fetch_annual_reports_sec.delay(symbol, count)
    return result.get(timeout=30)


@router.get("/10q/{symbol}")
@authenticate
def get_quarterly_reports(request, symbol: str, count: int = 5):
    """Get quarterly reports (10-Q)"""
    result = fetch_quarterly_reports_sec.delay(symbol, count)
    return result.get(timeout=30)


@router.get("/8k/{symbol}")
@authenticate
def get_current_reports(request, symbol: str, count: int = 10):
    """Get current reports (8-K) - material events"""
    result = fetch_current_reports_sec.delay(symbol, count)
    return result.get(timeout=30)


@router.get("/insider/{symbol}")
@authenticate
def get_insider_transactions(request, symbol: str, count: int = 50):
    """Get Form 4 insider trading filings"""
    result = fetch_insider_transactions_sec.delay(symbol, count)
    return result.get(timeout=30)


@router.get("/proxy/{symbol}")
@authenticate
def get_proxy_statements(request, symbol: str, count: int = 5):
    """Get proxy statements (DEF 14A)"""
    result = fetch_company_filings_sec.delay(symbol)
    return result.get(timeout=30)
